public/app/features/live/pages/RuleModal.tsx

```diff
@@ -1,23 +1,132 @@
-import React, { useState } from 'react';
-import { Modal, TabContent, TabsBar, Tab, CodeEditor } from '@grafana/ui';
-import { Rule } from './types';
-
+import React, { useState, useRef, useEffect } from 'react';
+import { Modal, TabContent, TabsBar, Tab, Button, Alert, useStyles } from '@grafana/ui';
+import { Rule, Setting, SettingLabel, PipelineListOption, EntitiesTypes } from './types';
+import { getBackendSrv } from '@grafana/runtime';
+import { css } from '@emotion/css';
+import { GrafanaTheme, SelectableValue } from '@grafana/data';
+import SettingsEditor from './SettingsEditor';
 interface Props {
   rule: Rule;
   isOpen: boolean;
   onClose: () => void;
+  clickColumn: Setting;
 }
-
-const tabs = [
+interface TabType {
+  label: SettingLabel;
+  value: Setting;
+}
+const tabs: TabType[] = [
   { label: 'Converter', value: 'converter' },
   { label: 'Processor', value: 'processor' },
   { label: 'Output', value: 'output' },
 ];
-const height = 600;
 
 export const RuleModal: React.FC<Props> = (props) => {
-  const { rule, isOpen, onClose } = props;
-  const [activeTab, setActiveTab] = useState<string>('converter');
+  const { rule, isOpen, onClose, clickColumn } = props;
+  // TODO: use reducer
+  const [activeTab, setActiveTab] = useState<Setting>(clickColumn);
+  const [success, setSuccess] = useState<boolean>();
+  const [error, setError] = useState<boolean>();
+  const [editedBody, setEditedBody] = useState<object | undefined>();
+  const [pipelineEntitiesList, setPipelineEntitiesList] = useState<any>();
+  const [pipelineEntitiesTypes, setPipelineEntitiesTypes] = useState<any>();
+  const [hasChange, setChange] = useState<boolean>(false);
+  const [currSetting, setCurrSetting] = useState<string>(clickColumn);
+  const [currSettingType, setCurrSettingType] = useState<any>('jsonAuto');
+  const isMounted = useRef(false);
+  const styles = useStyles(getStyles);
+
+  const onBlur = (text: string, setting: Setting, type: string | undefined) => {
+    setEditedBody(text ? JSON.parse(text) : undefined);
+    setChange(true);
+    setCurrSetting(setting);
+    setCurrSettingType(type);
+  };
+  useEffect(() => {
+    isMounted.current = true;
+    return () => {
+      isMounted.current = false;
+    };
+  }, []);
+  useEffect(() => {
+    getBackendSrv()
+      .get(`api/live/pipeline-entities`)
+      .then((data: EntitiesTypes) => {
+        if (isMounted) {
```

Comment:
is the point of this so we only do it once?  If so, should the wrapper be before making the HTTP call?